Plugin/Framework/Search/Request/AbstractHandler.php

```diff
@@ -272,27 +258,16 @@ abstract public function parseLimit(array $requestData);
      * @param $pageNum
      * @param $limit
      * @return array|null
+     * @throws NostoException
      */
     private function getCmpProductIds(FacetInterface $facet, $pageNum, $limit)
     {
-        try {
-            $res = $this->categoryService->getPersonalisationResult(
-                $facet,
-                $pageNum,
-                $limit
-            );
-            return $res ? CategoryMerchandising::parseProductIds($res) : null;
-        } catch (MissingCookieException $e) {
-            $this->logger->debugWithSource(
-                $e->getMessage(),
-                [],
-                $this
-            );
-            return null;
-        } catch (Exception $e) {
-            $this->logger->exception($e);
-            return null;
-        }
+        $res = $this->categoryService->getPersonalisationResult(
+            $facet,
+            $pageNum,
+            $limit
+        );
+        return $res ? CategoryMerchandising::parseProductIds($res) : null;
```

Comment:
@olsi-qose this method internally should log a warning when the result is not numeric. Allso, this methd should not return null.

Reply:
`$res` can take null value because `StateAwareCategoryService` can return null. Changing that would increase the scope and change.